config/confighttp/confighttp.go

```diff
@@ -227,9 +236,18 @@ func (hcs *ClientConfig) ToClient(ctx context.Context, host component.Host, sett
 		}
 	}
 
+	var jar http.CookieJar
+	if hcs.Cookies != nil && hcs.Cookies.Enabled {
+		jar, err = cookiejar.New(nil)
```

Comment:
Based on the docs of `cookiejar.Options`:

```
// PublicSuffixList is the public suffix list that determines whether
// an HTTP server can set a cookie for a domain.
//
// A nil value is valid and may be useful for testing but it is not
// secure: it means that the HTTP server for foo.co.uk can set a cookie
// for bar.co.uk.
```

and then it says

> A public suffix list implementation is in the package golang.org/x/net/publicsuffix.

Should we look into using https://pkg.go.dev/golang.org/x/net/publicsuffix ?

Reply:
ok I'll add that, good find thank you!